pkg/synthetictests/allowedbackenddisruption/matches.go

```diff
@@ -0,0 +1,89 @@
+package allowedbackenddisruption
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+// GetAllowedDisruption uses the backend and information about the clsuter to choose the best historical p95 to operate against.
+// We enforce "don't get worse" for disruption by watching the aggregate data in CI over many runs.
+func GetAllowedDisruption(ctx context.Context, backendName string, clientConfig *rest.Config) (*time.Duration, error) {
+	configClient, err := configclient.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	infrastructure, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	clusterVersion, err := configClient.ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	network, err := configClient.Networks().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	release := versionFromHistory(clusterVersion.Status.History[0])
+
+	fromRelease := ""
+	if len(clusterVersion.Status.History) > 1 {
+		fromRelease = versionFromHistory(clusterVersion.Status.History[1])
+	}
+
+	platform := strings.ToLower(string(infrastructure.Status.PlatformStatus.Type))
```

Comment:
I don't think this works on baremetal, which has a PlatformStatus.Type of BareMetal, but in BQ it's just `metal`:

<span ng-show="promiseResolved" ng-transclude="" aria-hidden="false" class="">

image-registry-reused-connections | 4.10 | 4.10 | metal | sdn | 2932.7 |  
-- | -- | -- | -- | -- | -- | --


</span>